cmd/manager/main.go

```diff
@@ -210,6 +212,7 @@ func addMetrics(ctx context.Context, cfg *rest.Config) {
 			log.Info("Install prometheus-operator in your cluster to create ServiceMonitor objects", "error", err.Error())
 		}
 	}
+	return err
```

Comment:
Having tested this out, what we're seeing is that if the metrics service fails to create (which is normal and expected in the case of the pod restarting) it causes the operator to fail.

So likely what needs to happen is:
- if the `serveCRMetrics` call fails and returns an error, this should pass the error back up the chain and cause the operator to restart.
- everything else in the function is fine to fail, as there are legitimate reasons for them to fail, so this `return err` at the end is not needed.